pages/api/endorsements/index.js

```diff
@@ -1,4 +1,38 @@
 // GET: /pages/api/endorsements/index.js
-export default function getTopEndorsements(req, res) {
-  // open to add query
+import { db } from '../../../utils/api/firebase.config';
+import {
+  collection,
+  query,
+  doc,
+  getDocs,
+  orderBy,
+  limit,
+} from "firebase/firestore";
+
+export default async function getTopEndorsements(req, res) {
+  try {
+    const q = query(collection(db, 'mentors'), orderBy('endorsements', 'desc'), limit(10));
+    const querySnapshot = await getDocs(q);
+
+    const result = [];
+    querySnapshot.forEach(doc => {
+      result.push(doc.data());
+      console.log(doc.data());
+    })
+
+    const transformResult = result.map(rank => {
+      return {
+        id: rank.id,
+        name: rank.name,
+        endorsements: rank.endorsements
+      }
+    });
```

Comment:
Better naming here too. Without the function name at the top I have very little idea what each variable name is supposed to be for.